libraries/browser-tracker-core/src/tracker/index.ts

```diff
@@ -652,6 +646,7 @@ export function Tracker(
       const idCookie = loadDomainUserIdCookie();
 
       domainUserId = initializeDomainUserId(idCookie, configAnonymousTracking);
+      memorizedVisitCount = visitCountFromIdCookie(idCookie);
```

Comment:
Shouldn't this go a few lines lower behind the if condition? The `startNewIdCookieSession()` function can change (increment) the visit count, so we should access the newer value.